### PR TITLE
Document performing insurance eligibility checks

### PR DESCRIPTION
**Pull request by ksmith94**
This will create documentation on how to perform an insurance eligibility check. Fixes issue #3204

### CODE REVIEW

**Comment by rahul1** [comment]
Additional video feedback: https://share.descript.com/view/8ySpcaoE4vg?transcript=false

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+| `unitPrice`        | The price charged to the patient for a single unit of the service. This is the price that the provider charges for the service.                                   |                                                         | $200                          |
+| `facility`         | A reference to the [`Location`](/docs/api/fhir/resources/location) or [`Organization`](/docs/api/fhir/resources/organization) where the service will be provided. |                                                         | Organization/example-hospital |
+| `detail`           | A reference to the [`CarePlan`](/docs/api/fhir/resources/careplan) with details describing the service.                                                           |                                                         | CarePlan/improve-vision       |
+
+<details><summary>Example: A coverage eligibility request for a basic consultation</summary>
```
Comment (rahul1) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:71]: We just upgraded to a newer version of Docusaurus which is more strict about MDX parsing. 

To get this to build, you should add the `<summary>` to a new line

Comment (rahul1) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:71]:
```suggestion
<details>
  <summary>Example: A coverage eligibility request for a basic consultation</summary>
```

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+# Insurance Eligibility Checks
+
+Insurance eligibility checks determine whether a patient's insurance is active, in-network, and has applicable benefits. They ensure that the provider will ultimately get compensated by the patient's insurer for a specific product or service.
+
+## Use Cases
+
+Insurance eligibility checks cover a variety of use cases, but three of the most important are:
+
+1. Requests for prior authorization
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:12]: Please put prior auth third on this list

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+2. Confirmation of benefits
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:13]: I'd describe this as Confirmation that insurance policy exists and is active

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+3. Coordination of care
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:14]: Perhaps: Confirm that insurance policy covers a specific [service type](https://x12.org/codes/service-type-codes)

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+### Requests for prior authorization
+
+Requests for prior authorization ensure that the patient is both covered and authorized to receive the requested care. This is used in cases where the insurer must ensure that the care is necessary before it is authorized. Often a patient must have a certain diagnosis or have attempted other care options prior to receiving authorization.
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:18]: requested care, for example a specific medication on procedure.

Comment (reshmakh) [reply packages/docs/docs/billing/insurance-eligibility-checks.md:18]: This should also go last IMO.

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+### Confirmation of benefits
+
+Insurance eligibility checks used for a confirmation of benefits are similar to requests for prior authorization, but they are simply checking whether the benefits cover the requested care. This is often for elective care and the check is to ensure that the care will be covered but does not require authorization from the insurer.
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:22]: Do you mean service types? https://x12.org/codes/service-type-codes - check whether this policy requires a given service type?

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+### Coordination of care
+
+In many cases patients can have multiple insurances policies, so eligibility checks are used to coordinate care. For example, if multiple policies cover the requested care, it needs to be determined which plan will actually pay for it. As the number of coverages and the products or services increase, this can become more complicated, so insurance eligibility checks are vital to make sure everything gets paid for.
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:26]: Not sure "coordination of care" is the right language here.  This is more like "coordinate coverage"

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+## Preparing an Eligibility Check
+
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:29]: Note first that the insurance cared and policy are represented by Coverage.  Link to coverage representation.

Comment (reshmakh) [reply packages/docs/docs/billing/insurance-eligibility-checks.md:29]: Nevermind, I see the descriptor below.

File: packages/examples/src/billing/insurance-eligibility-checks.ts [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        category: {
+          coding: [
+            {
+              system: 'http://terminology.hl7.org/CodeSystem/ex-benefitcategory',
+              code: '3',
+              display: 'Consultation',
+            },
+          ],
+        },
+        productOrService: {
```
Comment (reshmakh) [inline packages/examples/src/billing/insurance-eligibility-checks.ts:37]: Include X12? Candid, Change and others use X12 service codes https://developers.changehealthcare.com/eligibilityandclaims/docs/use-bare-minimum-eligibility-requests

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+| `provider`                | A reference to the [`Practitioner`](/docs/api/fhir/resources/practitioner) who is responsible for providing the service. |                 | Practitioner/dr-alice-smith              |
+| `category`                | The general type of the service or product being checked for eligibility.                                                |                 | Vision Coverage                          |
+
+:::note Multiple Insurances
+FHIR makes the `insurance` field on both the request and response an array, allowing for easy modeling of multiple insurance policies.
+
+When sending a request, if there are multiple insurances, the `CoverageEligibilityRequest.insurance.focal` field should be set to `true` on the specific coverage being checked.
+
+The `item` field is also an array on the `insurance` element of a [`CoverageEligibilityResponse`](/docs/api/fhir/resources/coverageeligibilityresponse) because it can represent multiple items that are covered under a specific insurance. When [coordinating care among multiple policies](#coordination-of-care), it can be common for multiple items from multiple coverages to be relevant to the check.
+
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:128]: Consider adding related reading:

Candid Health Eligibility check guide: https://docs.joincandidhealth.com/api-reference/eligibility
Change Healthcare Bare Minimum check: https://developers.changehealthcare.com/eligibilityandclaims/docs/use-bare-minimum-eligibility-requests

File: packages/docs/docs/billing/insurance-eligibility-checks.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
+import ExampleCode from '!!raw-loader!@site/..//examples/src/billing/insurance-eligibility-checks.ts';
+import MedplumCodeBlock from '@site/src/components/MedplumCodeBlock';
+
+# Insurance Eligibility Checks
+
+Insurance eligibility checks determine whether a patient's insurance is active, in-network, and has applicable benefits. They ensure that the provider will ultimately get compensated by the patient's insurer for a specific product or service.
+
```
Comment (reshmakh) [inline packages/docs/docs/billing/insurance-eligibility-checks.md:7]: This guide glosses over the most common eligibility check which will be is this policy active and does it cover `serviceType=30' (Plan Coverage and General Benefits
Plan coverage and general benefits for the member's policy or contract.) - this particular workflow should be discussed and maybe be made into an example.
https://developers.changehealthcare.com/eligibilityandclaims/docs/use-bare-minimum-eligibility-requests

Comment (ksmith94) [reply packages/docs/docs/billing/insurance-eligibility-checks.md:7]: I just pushed with updates that include this as a use case and provide examples. I also changed the use case section based on your suggestions.

**Comment by reshmakh** [comment]
Change Healthcare FAQ good resource: https://developers.changehealthcare.com/eligibilityandclaims/docs/eligibility-api-requests

**Review by rahul1 — APPROVED** [review]

**Pull request merged**